x/evm/types/utils.go

```diff
@@ -55,14 +55,17 @@ func DecodeTxResponses(in []byte) ([]*MsgEthereumTxResponse, error) {
 	if err := proto.Unmarshal(in, &txMsgData); err != nil {
 		return nil, err
 	}
-
-	responses := make([]*MsgEthereumTxResponse, len(txMsgData.MsgResponses))
-	for i, res := range txMsgData.MsgResponses {
+	responses := make([]*MsgEthereumTxResponse, 0)
+	for _, res := range txMsgData.MsgResponses {
 		var response MsgEthereumTxResponse
-		if err := proto.Unmarshal(res.Value, &response); err != nil {
+		if res.TypeUrl != "/"+proto.MessageName(&response) {
```

Comment:
do you mean a non-eth msg could be bundled together with a eth msg in a single tx?

Reply:
[yea](https://github.com/crypto-org-chain/cronos/actions/runs/6259692939/job/16998335986?pr=1163), and we have diff ibc proto in host and controller, and wrong wireType = 0 for field Hash trigger this bug

Reply:
I see, the tx is not necessarily a eth tx, when get logs from a block.